### ISSUE TITLE
Proposal: Top-5 circumplex-aware VIF critic improvements (with leakage controls)

### ISSUE THREAD

**Issue by DesmondChoy** [DesmondChoy/twinkl#26]
## Context

We are exploring how to improve the VIF critic by using Schwartz circumplex structure, especially **opposition relationships** (for example, Security vs Stimulation, Power vs Universalism).

<img width="375" height="266" alt="Image" src="https://github.com/user-attachments/assets/5231d178-d990-4aa3-a2ff-e62c04f7d9bb" />

Why this is worth doing now:

- Current critic runs still show neutral-collapse pressure and weak minority-class behavior in some dimensions.
- Some opposition pairs are visible in labels, but the model does not consistently encode these trade-offs.
- We optimize QWK/MAE/accuracy/calibration today, but we do not explicitly measure or optimize opposition-consistency.

This issue proposes the **top 5 implementation ideas** with the strongest impact-to-complexity ratio, plus leakage considerations.

---

### How it works
OVR measures how often canonical opposite pairs are predicted as same-direction extremes in one sample.

Example violation:
- `security > +0.6` and `stimulation > +0.6` (or both `< -0.6`)

This turns "circumplex inconsistency" into a measurable signal.

### Implementation plan
- Add `compute_opposition_violation_rate(...)` in `src/vif/eval.py`.
- Include OVR in outputs of `evaluate_model` and `evaluate_with_uncertainty`.
- Log OVR in `src/vif/experiment_logger.py` and notebook scorecards (`notebooks/critic_training/v2/critic_training_v2_nomic-embed.ipynb`).

### Data leakage considerations
- **Low risk** by itself.
- Leakage occurs only if we repeatedly tune on test OVR.
- Mitigation: tune on validation split only; reserve test for final one-shot report.

---

### How it works
Generate targeted synthetic entries centered on explicit opposite-value tensions (instead of generic value mentions), so the critic sees clearer supervision for trade-off dynamics.

### Implementation plan
- Add tension-pair config block in `config/synthetic_data.yaml`.
- Extend generation flow in `src/synthetic/generation.py` for tension-centered scenarios.
- Label generated entries with Judge pipeline and merge into training data.
- Keep existing banned-term/value-leakage protections intact.

### Data leakage considerations
- **Medium risk** if synthetic personas/near-duplicates leak across train/test.
- Mitigations:
  - Keep persona-level split discipline.
  - Freeze a test persona manifest.
  - Use train-only augmentation during iteration.
  - Continue banned-term enforcement.

---

### How it works
When Judge assigns strong non-zero to one value, it explicitly checks canonical opposite values for potential non-zero side-effects (while still allowing neutral if evidence is weak).

### Implementation plan
- Update `prompts/judge_alignment.yaml` with opposition-check guidance.
- Keep output schema unchanged for compatibility.
- Pilot relabel a subset, compare label distribution/agreement, then scale.

### Data leakage considerations
- **Medium risk** if prompt versions are tuned based on test outcomes.
- Mitigations:
  - Maintain a prompt-dev holdout set.
  - Freeze prompt version before final evaluation.
  - Version labels by prompt version.

---

### How it works
Upweight training samples that contain non-neutral opposite-pair signal so mini-batches are less neutral-dominated and expose the model to more informative tension examples.

### Implementation plan
- Compute train-only sample weights from labels.
- Use `WeightedRandomSampler` in training DataLoader (`src/vif/dataset.py` or v2 notebook pipeline).
- Keep validation/test loaders unchanged.

### Data leakage considerations
- **Low risk** if weights are computed only from training split.
- Leakage risk if val/test labels influence sampler stats.
- Mitigation: strict split isolation for all sampler statistics.

---

### How it works
Add a soft penalty when opposite pairs strongly co-activate with same sign.

Objective:
`L_total = L_base + lambda_opp * L_opp`

where `L_opp` penalizes high same-sign co-activation for canonical opposite pairs.

### Implementation plan
- Add helper(s) in `src/vif/critic_ordinal.py` to compute continuous expected per-dimension scores.
- Add opposition penalty term in notebook training loop (`critic_training_v2_nomic-embed.ipynb`) with ablations over `lambda_opp` and margin.
- Start with small `lambda_opp` to avoid over-constraining valid edge cases.

### Data leakage considerations
- **Low risk** when using theory-defined canonical pairs.
- Leakage risk if pair strengths are estimated from full dataset including test.
- Mitigation: fixed theory pairs or train-only estimation.

---

## Success criteria (proposed)

- QWK improves over current baseline while calibration does not regress.
- Minority-class recall improves (especially class `-1` and `+1`).
- Hedging/neutral-collapse decreases.
- OVR decreases without over-suppressing predictions.

---

## Suggested rollout order

1. OVR metric
2. Opposition-aware batch sampling
3. Soft opposite regularizer
4. Judge rubric upgrade
5. Opposite-aware synthetic augmentation

This ordering gives observability first, then low-risk model-side wins, then label/data upgrades.
